crates/turborepo-lib/src/shim.rs

```diff
@@ -74,6 +76,13 @@ impl ShimArgs {
             })
         }
     }
+
+    // returns true if any flags result in pure json output to stdout
+    pub fn has_json_flags(&self) -> bool {
+        self.remaining_turbo_args
+            .iter()
+            .any(|arg| arg == "--json" || arg == "--dry=json" || arg == "--graph")
```

Comment:
There are some additional ways to invoke this:

```
"--dry=json"
"--dry-run=json"
"--dry json"
"--dry-run json"
```

I'm not 100% sure if this is a full enumeration.